universe/Meter.cpp

```diff
@@ -10,40 +10,46 @@
 #endif
 
 namespace {
+    consteval bool fabs_close_enough(float f1, float f2) {
```

Comment:
Have you considered using some test frameworks?   

Usually, they come with human-friendly output, shorter code, and float comparison support.

Reply:
Not in this context... I don't know what a "framework" is or could provide that `static_assert` doesn't. This is compile time checking that low-level operations work as expected and that types are defined as expected. I don't need a more "human-friendly" output for that, and if I did, I could add a custom error message about it to the `static_assert`.

What does "float comparison support" mean in practice?

There is already a bunch of test code for systems level runtime stuff (eg. parse, start a game and run a turn) that uses Boost Test. That's not like what this code is doing.

Reply:
Framework is responsible for collecting/running and reporting tests, like Boost Test.    I see that this is other type of tests. 


> What does "float comparison support" mean in practice?

Someone wrote this comparison function for you. For example [pytest has one](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-approx).  Sometimes it has warnings, that you compare float with simple equal.

Reply:
In this case I specifically wanted to test that the results were exactly equal, to ensure that they were consistent across systems. It's not a matter of a calculation giving slightly different results than would a more-precisely propagated version of the same calculation.

Reply:
> In this case I specifically wanted to test that the results were exactly equal, to ensure that they were consistent across systems

I might be wrong, but usually, float arithmetic depends more on the hardware than on the software.  But since we don't support various devices, this should be ok.